frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx

```diff
@@ -0,0 +1,137 @@
+import userEvent from "@testing-library/user-event";
+import { Route } from "react-router";
+import { mockSettings } from "__support__/settings";
+import { renderWithProviders, screen, within } from "__support__/ui";
+import { createMockState } from "metabase-types/store/mocks";
+import { Button, Group, Text } from "metabase/ui";
+import { EmbedModal } from "./EmbedModal";
+
+// We need to mock the Modal here because there's an issue with the RoutelessFullPageModal
+// component, which can't find the routing.locationBeforeTransitions prop when it's rendered with this test.
+// We'll just take the props that the Modal is using and render them so we know what the component is
+// working with, and then we can refactor this test later using the Mantine Modal when all inner
+// components are refactored to use Mantine.
+jest.mock("metabase/components/Modal", () => {
```

Comment:
let me know if there's a better way to handle this, I'm not exactly a fan

Reply:
I'd not set `full: true` because in [milestone 5 ](https://github.com/metabase/metabase/pull/36181), the modal would not be full anyway. This prevents `Modal` from rendering `RoutelessFullPageModal` which causes the error you mentioned.

This might be worth the compromise, but let's discuss with Product too.

You need to play around with the modal size a bit. In this example I pass `wide: true` and `fit: embedType !== "application"` to `Modal` component in `EmbedModal`

![image](https://github.com/metabase/metabase/assets/1937582/968a6eec-92be-4d85-a7af-cb62ac5aaa02)

Otherwise, if we decide to keep this mock, let's add a comment to ensure we won't forget to revisit this after milestone 5 since we won't render `RoutelessFullPageModal` in embed modal anymore.

Reply:
Yeah, I was thinking about that as well. My thinking was that I should try and reduce the amount of user-facing changes in this milestone since I know Denis is going to move things around, so I wanted to keep things relatively predictable for him

Reply:
I like this huge modal style for the time-being so we can reduce the code size, since Milestone 5 is coming very soon 😄

Reply:
Nice! So I used your suggestion and I could immediately remove all of the mock code and it still looks good 😄